### PR TITLE
qa: check each fs for health

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/59425

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/tasks/ceph.py [hunk 1]
```diff
@@ -1495,9 +1495,8 @@ def healthy(ctx, config):
 
     if ctx.cluster.only(teuthology.is_type('mds', cluster_name)).remotes:
         # Some MDSs exist, wait for them to be healthy
-        ceph_fs = Filesystem(ctx) # TODO: make Filesystem cluster-aware
-        ceph_fs.wait_for_daemons(timeout=300)
-
+        for fs in Filesystem.get_all_fs(ctx):
```
Comment (joscollin) [inline qa/tasks/ceph.py:1498]: Why don't just call `FSStatus.get_filesystems()`? either by importing `FSStatus` or by calling `ceph_fs.status()`.
Any issue in that?

**Comment by batrick** [comment]
jenkins test windows

**Comment by batrick** [comment]
jenkins test windows

**Comment by vshankar** [comment]
jenkins test windows

**Comment by dparmar18** [comment]
maybe add "`caused by` https://github.com/ceph/ceph/pull/50896/commits/69565dbbf124e29f30dae989eae41eada161a4b4" in commit message?

**Comment by vshankar** [comment]
@rishabh-d-dave This fixes failures like: https://pulpito.ceph.com/yuriw-2023-04-18_14:39:18-fs-wip-yuri6-testing-2023-04-12-0732-reef-distro-default-smithi/7244929

(testing in main branch, but the test run has other failures - so did not record the run in run log).

**Comment by rishabh-d-dave** [comment]
As per my conversation with @vshankar, I am merging this PR and backporting it since this PR fixes lots of failures that are being seen in Pacfic and Reef backport QA runs.

**Pull request merged**